packages/gdpr-tool/src/main.js

```diff
@@ -0,0 +1,215 @@
+    return Promise.all(
+      getUris.map(uri => {
+        const request = GDPRTool.buildRequest(uri, 'GET')
+
+        return this.client.execute(request)
+      })
+    )
+      .then(async data => {
+        let results = flatten(data.map(response => response.body.results))
+        const ids = results.map(result => result.id)
+
+        if (ids.length > 0) {
+          const reference = GDPRTool.buildReference(ids)
+
+          const messagesUri = requestBuilder.messages
+            .where(reference)
+            .perPage(500)
+            .build()
+
+          const request = GDPRTool.buildRequest(messagesUri, 'GET')
+          const messages = await this.client.execute(request)
+
+          results = [...messages.body.results, ...results]
+        }
+        return Promise.resolve(results)
+      })
+      .catch(error => Promise.reject(error))
+  }
+
+  deleteData(customerId) {
+    if (!customerId) throw Error('missing `customerId` argument')
+    this.logger.info('Starting deletion')
+
+    const requestBuilder = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const customersUri = {
+      uri: requestBuilder.customers
+        .where(`id = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.customers,
+    }
+    const ordersUri = {
+      uri: requestBuilder.orders
+        .where(`customerId = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.orders,
+    }
+    const cartsUri = {
+      uri: requestBuilder.carts
+        .where(`customerId = "${customerId}"`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.carts,
+    }
+    const paymentsUri = {
+      uri: requestBuilder.payments
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.payments,
+    }
+    const shoppingsListsUri = {
+      uri: requestBuilder.shoppingLists
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.shoppingLists,
+    }
+    const reviewsUri = {
+      uri: requestBuilder.reviews
+        .where(`customer(id = "${customerId}")`)
+        .perPage(500)
+        .build(),
+      builder: requestBuilder.reviews,
+    }
+
+    const getUris = [
+      customersUri,
+      ordersUri,
+      cartsUri,
+      paymentsUri,
+      shoppingsListsUri,
+      reviewsUri,
+    ]
+
+    // todo change to process instead of execute
+    // in case of more than 500 results
+    getUris.forEach(uri => {
+      const request = GDPRTool.buildRequest(uri.uri, 'GET')
+
+      this.client
+        .execute(request)
+        .then(async payload => {
+          if (payload.statusCode !== 200) return Promise.reject(payload)
+          GDPRTool.delete(payload, uri.builder)
+
+          return Promise.resolve()
+        })
+        .catch(error => Promise.reject(error))
```

Comment:
Why do you need to catch and reject the same error again? Can't we just omit the catch block in this case?

Reply:
Ahh thanks a lot, I didnt think of that 👍